api/src/migrations/geoprocessing/1611221157285-initialGeoDBSetup.ts

```diff
@@ -0,0 +1,185 @@
+        'done',
+        'failure'
+      );
+
+      CREATE TYPE "adm_level" AS ENUM (
+        0,
+        1,
+        2
+      );
+
+-- Administrative regions table.
+      CREATE TABLE "admin_regions" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "ogc_fid" int NOT NULL,
+        "the_geom" geometry CONSTRAINT admin_regions_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "name_0" varchar,
+        "name_1" varchar,
+        "name_2" varchar,
+        "iso3" varchar(3),
+        "gid_0" varchar,
+        "gid_1" varchar,
+        "gid_2" varchar,
+        "level" adm_level NOT NULL,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      ) PARTITION BY LIST(level);
+
+      CREATE TABLE admin_regions_0 PARTITION OF admin_regions FOR VALUES IN (0);
+      CREATE TABLE admin_regions_1 PARTITION OF admin_regions FOR VALUES IN (1);
+      CREATE TABLE admin_regions_2 PARTITION OF admin_regions FOR VALUES IN (2);
+
+      -- Protected Areas table.
+      CREATE TABLE "wdpa" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "wdpaid" float8,
+        "the_geom" geometry CONSTRAINT wdpa_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "full_name" varchar,
+        "iucn_cat" varchar,
+        "shape_leng" float8,
+        "shape_area" float8,
+        "iso3" varchar(3) REFERENCES "admin_regions" ("iso3"),
+        "status" text,
+        "desig" text,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Features data table (will hold species and bioma data).
+      CREATE TABLE "features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "properties" jsonb,
+        "source" source_type
+      );
+
+      -- Usage of features data within scenarios table (will hold species and bioma data).
+      CREATE TABLE "scenario_features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "feature_class_id" uuid,
+        "scenario_id" uuid,
+        "total_area" varchar,
+        "admin_area" timestamp,
+        "current_pa" source_type,
+        "spf" float8,
+        "target" float8,
+        "target2" float8,
+        "targetocc" float8,
+        "sepnum" float8,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Planning units geometry data table.
+      CREATE TABLE "planing_units_geom" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry UNIQUE NOT NULL CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "type" shape_type,
+        "size" int
```

Comment:
just a very mild suggestion - unless it's very clear what the measurement unit is, I find it useful to add it in the name for very generic column names such as `size`

Reply:
I see there's a comment for this, but (very personal preference!) I would just rename this `size_in_km2` and avoid the comment